src/workers/generateAsync.worker.js

```diff
@@ -43,6 +43,12 @@ global.onmessage = function ( { data } ) {
 
 		}
 
+		if ( options.groups ) {
+
+			geometry.groups = options.groups;
+
+		}
```

Comment:
Can we iterate over the geometry groups here and use the [geometry.addGroup](https://threejs.org/docs/?q=buffergeometry#api/en/core/BufferGeometry.addGroup) function here, instead? If the function changes, is deprecated, or has side effects added later on it would be best to go through the correct method.

Reply:
ok, Ill try and have a look later when Im home.
Didnt know about the function 😂
Im lazy in my experiment and just added what I needed.

Noticed the material index, can we actually have different materials on merged meshes by using the groups ? Do you know ?

Reply:
btw, tested the material index part... we could have own material per group... but ended up with own draw call per group if I did :-(

Reply:
This is more of a three.js question rather than a bvh question but if you have multiple groups on a mesh then they are always rendered with a different draw call for each one even if the same material is used.